### ISSUE TITLE
Recovery processing way too fast?

### ISSUE THREAD

**Issue by ivplaton** [gurnec/btcrecover#82]
I am trying to recover a lost Electrum 2.x wallet password. 

After compiling a list of possible combinations, it comes up to some 4 million, which would be about right. 

However, it then somehow manages to try and "exhaust" all of these in some 30 - 60 seconds. Which makes me think of some possible issue on my side, since I'm running the script on a 2 years old laptop.

Any idea on what this could be caused by? Have somebody else experienced this before?

**Comment by gurnec** [comment]
This sounds normal to me. With most<sup>*</sup> Electrum wallets, btcrecover takes very little time to check passwords because Electrum doesn't use any [key stretching](https://en.wikipedia.org/wiki/Key_stretching). There are some brute-forcing tools out there that can run even faster (10x faster).

You can run (just double-click it) the `run-all-tests.py` script to ensure there's nothing unusual going on with your system (skipped tests are to be expected, as long as there are no failed tests).

<sub>* Electrum 2.8 can optionally (it's off by default) create fully-encrypted wallet files which do use some minimal key stretching.</sub>

**Comment by ivplaton** [comment]
Thanks for your input! 

I was just surprised by the speed of it (and, of course, by the fact that I wasn't right with any of the 4 mil combinations).

run-all-tests.py returns OK, only skipping stuff related to the GPU addon which I understand isn't useful for Electrum anyway?

Well, back to memory digging, then.

**Comment by gurnec** [comment]
> the GPU addon which I understand isn't useful for Electrum anyway?

Yup, as far as I know there doesn't exist a tool for using GPUs against Electrum wallets. Due to Electrum's lack of of key stretching, GPUs probably aren't a whole lot better than CPUs anyways.

If your password is short enough, you may want to check out a [brute-forcer such as JohnTheRipper](https://github.com/magnumripper/JohnTheRipper) instead. Using C instead of Python provides a big speed boost, although that tool is more for brute-forcing when you've no idea what the password is (and therefore it has to check many more possibilities). You'll also need to compile it yourself.

> Well, back to memory digging, then.

'Fraid so, unless you have the seed somewhere of course... good luck!

**Comment by nori19870522** [comment]
I am Japanese and do password analysis.
https://keychainx.io/

This is Robert's article.
https://news.bitstarz.com/bitstarz-news-interview-robert-rhodin-keychainx-ceo

I may be able to find the password you lost. Do you need help with your login?

**Comment by btcexchanging** [comment]
if you haven't recovered your wallet, i can help you recover your wallet.
I have recovered most of the common wallets and recovery phrases for many users
you can reach me here at my telegram account:  @walletrecover

**Comment by viladmir2022** [comment]
hi, if you still have wallet problem. I can recover your lost wallet.
I have powerful servers and excellent software that can find any password.
and my fee is only 15-25% of the recovered fund
here is my telegram id: @viladmir2022 and Gmail: walletsrecoveryservice@gmail.com
